cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -17,6 +17,8 @@
 
 #include "arrow/filesystem/azurefs.h"
 
+#include <azure/identity/default_azure_credential.hpp>
+#include <azure/storage/blobs.hpp>
 #include "arrow/result.h"
```

Comment:
Nit: please insert a newline between in-project includes and third-party includes.
```suggestion

#include "arrow/result.h"
```